### PR TITLE
Use mwc-drawer (#10335

### PR DESCRIPTION
**Pull request by bramkragten**
## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by balloob** [comment]
I've talked to both Android and iOS:

Android wants to get rid of the swipe from the side gesture because it is no longer part of the Android guidelines (instead, swipe from side = back).

iOS is going to implement their own sidebar. I've added #11347 to unblock this PR.

**Comment by LasseRosenow** [comment]
> I've talked to both Android and iOS:
> 
> Android wants to get rid of the swipe from the side gesture because it is no longer part of the Android guidelines (instead, swipe from side = back).
> 
> iOS is going to implement their own sidebar. I've added #11347 to unblock this PR.

As far as I know, on Android the swipe from the left is still possible on the top left part of the phone

**Comment by LasseRosenow** [comment]
> > I've talked to both Android and iOS:
> > 
> > Android wants to get rid of the swipe from the side gesture because it is no longer part of the Android guidelines (instead, swipe from side = back).
> > 
> > iOS is going to implement their own sidebar. I've added #11347 to unblock this PR.
> 
> As far as I know, on Android the swipe from the left is still possible on the top left part of the phone

As it seems this behavior differs on Android versions, or maybe on vender specific stuff ..

It is not working on vanilla android 12 anymore it seems, so sorry for the confusion.


But we should maybe keep in mind, that the gesture still works on all devices that use legacy navigations and don't habe gesture navigation enabled.

Also it is possible to Trag from the side in some sort of ange downwards. This does not trigger the back gesture, but opens the drawer

Also it is possible do open the drawer by putting your thumb on the left hand corner and wait for a second. It the appears and can be opened by dragging. (This is what I always do)

Maybe the Android App should implement it's own drawer as well, to handle all these pain points 🤔

**Comment by balloob** [comment]
Dragging diagnoally to open the menu is not a feature I think many people will even try, because if you mess up you're doing a back gesture.

For legacy devices it's a feature that we will drop but that's an acceptable compromise, as it will greatly improve our loading speed as we can finally drop Polymer from the hot path.

**Comment by LasseRosenow** [comment]
I am not so sure about as to how `legacy` 3-button navigation is, as you can still enable it on modern android and even some phones like my xiaomi mi 11 have it enabled by default and I had to switch to gesture myself.

I think too, that moving to the more modern web component is very good, but I was thinking that maybe it could be somewhat extended to address those issues? 🤔 

Especially being able to open the drawer while putting the thumb on the left and waiting for a short time is something that would be very nice to have, because all other android apps do it and otherwise navigation with one hand is not really possible anymore.

But if it is not possible to extend the modern component, I think implementing a custom drawer on android would do the job as well and even improve on current behaviour. I think I would be very much in favor of this approach.

**Comment by balloob** [comment]
Ideally we would add that to the Android app and send a request to the UI to open the sidebar

File: src/html/_style_base.html.template [hunk 1]
```diff
@@ -1,5 +1,8 @@
 <meta name='viewport' content='width=device-width, user-scalable=no, viewport-fit=cover, initial-scale=1'>
 <style>
+  html {
+    overflow: hidden;
+  }
```
Comment (piitaya) [inline src/html/_style_base.html.template:5]: Is this needed ?

**Review by piitaya — APPROVED** [review]

**Review by piitaya — CHANGES_REQUESTED** [review]
The developer tools header seems to be broken

**Pull request merged**